src/Compiler/Microsoft.AspNetCore.Razor.Language/test/Legacy/CSharpBlockTest.cs

```diff
@@ -742,6 +754,111 @@ public void WithUnexpectedTransitionsInAttributeValue_Throws()
         ParseDocumentTest("@{<span foo='@ @' />}");
     }
 
+    [Fact]
+    public void EscapedIdentifiers_01()
+    {
+        ParseDocumentTest("""
+            @{
+                var @string = "blah";
+            }
+
+            @(@string)
+            """);
+    }
+
+    [Fact]
+    public void EscapedIdentifiers_02()
+    {
+        ParseDocumentTest("""
+            @{
+                @string.Format("1{0}", DateTime.Now)
+                var x = 1;
+                var y = @x;
+                @string.Format("2{0}", DateTime.Now)
+            }
+            """);
+    }
+
+    [Fact]
+    public void EscapedIdentifiers_03()
+    {
+        ParseDocumentTest("""
+            @{
+                var @@class = 1;
+                var y = @@class;
+            }
+            """);
+    }
+
+    [Fact]
+    public void EscapedIdentifiers_04()
+    {
+        ParseDocumentTest("""
+            @{
+                var @string = "string test";
+                @string = "new string";
```

Comment:
Consider adding codegen test with some razor like

```razor
@{
    var x = "hello";
    @x x = "world"; @x
}
```

that should render "helloworld" right?

My point is that we should test this "feature" where you can render values from within a code block and then continue writing C# statements after that rendered value on the same line. Unless we already have such a test, that's also likely.